packages/zoe/src/contractSupport/ratio.js

```diff
@@ -61,32 +65,28 @@ export const makeRatio = (
     X`No infinite ratios! Denominator was 0/${q(denominatorBrand)}`,
   );
 
-  // TODO(https://github.com/Agoric/agoric-sdk/pull/2310) after the refactoring
-  //  use AmountMath's constructor here rather than building the record directly
   return harden({
-    numerator: { value: Nat(numerator), brand: numeratorBrand },
-    denominator: { value: Nat(denominator), brand: denominatorBrand },
+    numerator: AmountMath.make(numeratorBrand, Nat(numerator)),
+    denominator: AmountMath.make(denominatorBrand, Nat(denominator)),
   });
 };
 
+/** @type {MakeRatioFromAmounts} */
 export const makeRatioFromAmounts = (numeratorAmount, denominatorAmount) => {
-  // TODO(https://github.com/Agoric/agoric-sdk/pull/2310) after the refactoring
-  // coerce amounts using a native AmountMath operation.
-
+  AmountMath.coerce(numeratorAmount.brand, numeratorAmount);
+  AmountMath.coerce(denominatorAmount.brand, denominatorAmount);
   return makeRatio(
+    // @ts-ignore coerce ensures values are Nats
     Nat(numeratorAmount.value),
```

Comment:
See if this works rather than ts-ignore?

```suggestion
    Nat(/** @type {NatValue} */ (numeratorAmount.value)),
```

Reply:
Yep, that works. Thanks!